The 'stop' argument causes 'pyo3_runtime.PanicException' in some cases.

**The bug**
Hello, Thanks for your great work. But recently I have occured some wired bugs and don't how to fix it.
I want to gen a json format string using `gen` with some regular args.
**Using the `stop` argument in `gen` can cause bugs in certain cases.**

**Code** 
```python
wmeta_key = wordmeta.get_attr_name_list()   #Basemodel from pydantic
    input_text = f"task:{task}\nprompt:{prompt}\n```json\n{{\n"
    for key in wmeta_key:
        wmeta_value = getattr(wordmeta, key)
        wdecision_value = getattr(wdecision, f"need_{key}")

        if isinstance(wmeta_value, int | float):
            reg = r"^\d+$"
            m_tokens = 10
        elif isinstance(wmeta_value, str):
            reg = None
            m_tokens = 50

        gen_res = gen_decision(
            lm + f"{task} {key}: ",
            need_llm=wdecision_value,
            ori_text=wmeta_value,
            name="res",
            regex=reg,
            stop=["'", '"', ".", "\n"],
            max_tokens=m_tokens,
        )

def gen_decision(lm, need_llm: bool, ori_text: str, **gen_args):
    return lm + gen(**gen_args) if need_llm else ori_text

......
``` 

When running `gen_decision`, error will be reported. If I remove the `stop` argument, the issue does not occur. However, the output generated by the LLM is not satisfactory.
```
assertion failed: self.state.byte_to_token_idx.len() >= n_bytes
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "/work/llm.py", line 65, in gen_decision
    return lm + gen(**gen_args) if need_llm else ori_text
  File "/root/anaconda3/envs/guidance/lib/python3.10/site-packages/guidance/models/_model.py", line 1207, in __add__
    out = lm._run_stateless(value)
  File "/root/anaconda3/envs/guidance/lib/python3.10/site-packages/guidance/models/_model.py", line 1413, in _run_stateless
    for chunk in gen_obj:
  File "/root/anaconda3/envs/guidance/lib/python3.10/site-packages/guidance/models/_model.py", line 431, in __call__
    tokens, mask_fut, backtrack = parser.advance(engine_output)
  File "/root/anaconda3/envs/guidance/lib/python3.10/site-packages/guidance/_parser.py", line 78, in advance
    return self._generator.send(engine_output)
  File "/root/anaconda3/envs/guidance/lib/python3.10/site-packages/guidance/_parser.py", line 153, in _parse
    backtrack, ff_tokens = self.ll_interpreter.commit_token(
pyo3_runtime.PanicException: assertion failed: self.state.byte_to_token_idx.len() >= n_bytes
[ERROR] 2025-02-21-07:57:02 (PID:3017059, Device:0, RankID:-1) ERR99999 UNKNOWN application exception
```

But when running some simple codes, it will be ok:
```python
lm = create_lm() #lm = models.Transformers(model_path, device_map=device_map)
    print(
        lm
        + "Hello "
        + gen(
            name="res",
            regex=None,
            stop=["'", '"', ";", ":", ",", ".", "\n"],
            max_tokens=50,
            temperature=0.5,
        )
    )

```
Response
```
Hello I am trying to create a simple program that will take a string and print out the number of times each character appears in the string
```

**To Reproduce**
Loader :  loaded by `models.Transformers`
Model: `Qwen2.5-1.5b-instruct`, `Qwen1.5-7b`
Executing the aforementioned code

**System info (please complete the following information):**
 - OS (e.g. Ubuntu, Windows 11, Mac OS, etc.): Ubuntu
 - Guidance Version (`guidance.__version__`):  0.2.0
 - Device: npu

--- a/setup.py
+++ b/setup.py
@@ -34,7 +34,7 @@ install_requires = [
     "psutil",
     "tiktoken>=0.3",
     "guidance-stitch",
-    "llguidance==0.6.30",
+    "llguidance==0.6.31",
     "setuptools" # TODO - Remove before release, used for multimodal mocks in python 3.12
 ]
 

